tfhe/src/integer/server_key/radix_parallel/scalar_comparison.rs

```diff
@@ -638,52 +636,503 @@ impl ServerKey {
+                ComparisonKind::GreaterOrEqual => {
+                    let mut sign_bit = if self.message_modulus().0 > 2 {
+                        let sign_bit_lut = self.key.generate_lookup_table(|last_block| {
+                            let modulus = self.key.message_modulus.0 as u64;
+                            (last_block % modulus) / (modulus / 2)
+                        });
+                        let sign_bit = self
+                            .key
+                            .apply_lookup_table(a.blocks().last().unwrap(), &sign_bit_lut);
+                        BooleanBlock::new_unchecked(sign_bit)
+                    } else {
+                        BooleanBlock::new_unchecked(a.blocks().last().cloned().unwrap())
+                    };
+                    self.boolean_bitnot_assign(&mut sign_bit);
+                    return sign_bit;
+                }
+                ComparisonKind::LessOrEqual | ComparisonKind::Greater => {}
+            }
+        } else if !T::IS_SIGNED && b == Scalar::ZERO {
+            match compare {
+                ComparisonKind::Less => return self.create_trivial_boolean_block(false),
+                ComparisonKind::GreaterOrEqual => return self.create_trivial_boolean_block(true),
+                ComparisonKind::LessOrEqual | ComparisonKind::Greater => {}
+            }
+        }
+
+        let packed_modulus = (self.key.message_modulus.0 * self.key.message_modulus.0) as u64;
+
+        // We have that `a < b` <=> `does_sub_overflows(a, b)` and we know how to do this.
+        // Now, to have other comparisons, we will re-express them as less than (`<`)
+        // with some potential boolean negation
+        //
+        // Note that for signed ciphertext it's not the overflowing sub that is used,
+        // but it's still something that is based on the subtraction
+        //
+        // For both signed and unsigned, a subtraction with borrow is used
+        // (as opposed to adding the negation)
+        let num_block_is_even = (a.blocks().len() & 1) == 0;
+        let a = a
+            .blocks()
+            .chunks(2)
+            .map(|chunk_of_two| self.pack_block_chunk(chunk_of_two))
+            .collect::<Vec<_>>();
+
+        let padding_value = (packed_modulus - 1) * u64::from(b < Scalar::ZERO);
+        let mut b_blocks = BlockDecomposer::new(b, packed_modulus.ilog2())
+            .iter_as::<u64>()
+            .chain(std::iter::repeat(padding_value))
+            .take(a.len())
+            .collect::<Vec<_>>();
+
+        if !num_block_is_even && b < Scalar::ZERO {
+            let last_index = b_blocks.len() - 1;
+            // We blindly padded with the ones, but as the num block is not even
+            // the last packed block high part shall be 0 not 1s (i.e. no padding)
+            b_blocks[last_index] %= self.message_modulus().0 as u64;
+        }
+
+        let b = b_blocks;
+        let block_modulus = self.message_modulus().0 * self.carry_modulus().0;
```

Comment:
this seems to be equivalent to packed_modolus, so maybe you can reuse this variable